errcheck/creator/creator.go

```diff
@@ -52,7 +70,35 @@ func Errcheck() checker.Creator {
 						}
 						return okgo.NewIssueFromLine(line, wd)
 					},
-				)), nil
+				),
+			)
+			return &cleanupChecker{c, cleanup}, nil
 		},
 	)
 }
+
+type cleanupChecker struct {
+	okgo.Checker
+	cleanup func()
+}
+
+func (c *cleanupChecker) Check(pkgPaths []string, projectDir string, stdout io.Writer) {
+	if c.cleanup != nil {
+		defer c.cleanup()
+	}
+	c.Checker.Check(pkgPaths, projectDir, stdout)
+}
+
+func createExcludeFile(excludes []string) (string, error) {
+	tmpExcludeFile, err := ioutil.TempFile("", "")
+	if err != nil {
+		return "", errors.Wrap(err, "failed to create temporary file")
+	}
+	if _, err := tmpExcludeFile.WriteString(strings.Join(excludes, "\n")); err != nil {
+		return "", errors.Wrap(err, "failed to write excludes to file")
+	}
+	if err := tmpExcludeFile.Close(); err != nil {
```

Comment:
The `close` call should be deferred after the `ioutil.TempFile` call so that the file will be closed if the return at line 98 is hit. I would use the pattern of having named return values and setting the returned error to be the error returned by `Close()` if it is non-nil.